control-plane/api-gateway/binding/binder.go

```diff
@@ -219,9 +231,38 @@ func (b *Binder) Snapshot() Snapshot {
 		entry := b.config.Translator.GatewayToAPIGateway(b.config.Gateway, seenCerts)
 		snapshot.Consul.Updates = append(snapshot.Consul.Updates, &entry)
 
+		registrationPods := []corev1.Pod{}
+		// filter out any pod that is being deleted
+		for _, pod := range b.config.Pods {
+			if !isDeleted(&pod) {
+				registrationPods = append(registrationPods, pod)
+			}
+		}
+
+		registrations := registrationsForPods(entry.Namespace, b.config.Gateway, registrationPods)
+		snapshot.Consul.Registrations = registrations
+
+		// deregister any not explicitly registered service
+		for _, service := range b.config.GatewayServices {
+			found := false
+			for _, registration := range registrations {
+				if service.ServiceID == registration.Service.ID {
+					found = true
```

Comment:
should we break out of this loop here once we find the service?

Reply:
Yeah, we could do the early break, just forgot to add it. Since this is so big though, think we can probably just add it in later as we identify bug fixes.